### PR TITLE
[MM-29311] Migrate string-refs: this.refs.rhspostlist

### PR DESCRIPTION
**Pull request by mathiasvr**
#### Summary

Migrate string-refs to functional ones: `this.refs.rhspostlist`

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/15800
https://mattermost.atlassian.net/browse/MM-29311

### CODE REVIEW

File: components/rhs_thread/rhs_thread.tsx [hunk 1]
```diff
@@ -97,6 +98,8 @@ export default class RhsThread extends React.Component<Props, State> {
             topRhsPostId: '',
             openTime,
         };
+
+        this.rhspostlist = React.createRef();
```
Comment (sudheerDev) [inline components/rhs_thread/rhs_thread.tsx:102]: Can we rename this to `this.rhspostlistRef`, to be consistent with other instances and PR's

**Review by sudheerDev — CHANGES_REQUESTED** [review]

File: components/rhs_thread/rhs_thread.tsx (hunk 1 above)
Comment (mathiasvr) [reply components/rhs_thread/rhs_thread.tsx:102]: Sure

**Review by sudheerDev — APPROVED** [review]
LGTM

**Comment by sudheerDev** [comment]
@mathiasvr Thank you 🎉

**Review by Willyfrog — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @mathiasvr 
Tested, looks good to merge.
- Verified/tested around RHS text box - as expected.

**Comment by Willyfrog** [comment]
fixes https://github.com/mattermost/mattermost-server/issues/15800

**Pull request merged**